java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -1087,6 +1087,24 @@ public void smove(BaseClient client) {
         assertInstanceOf(RequestException.class, executionException.getCause());
     }
 
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void rename(BaseClient client) {
+        String key1 = "{key}" + UUID.randomUUID();
+
+        assertEquals("OK", client.set(key1, "foo").get());
+        assertEquals("OK", client.rename(key1, key1 + "_rename").get());
+        assertEquals(1L, client.exists(new String[] {key1 + "_rename"}).get());
+
+        // key doesn't exist
+        ExecutionException executionException =
+                assertThrows(
+                        ExecutionException.class,
+                        () ->
+                                client.rename("{same_slot}" + "non_existing_key", "{same_slot}" + "_rename").get());
+    }
+
```

Comment:
Since this command is a multi-key command that does not allow passing keys with different hash tags, you will also have to add a line to cluster/CommandTests.java#callCrossSlotCommandsWhichShouldFail

Reply:
^